openfl/component/collaborator/collaborator.py

```diff
@@ -162,6 +188,14 @@ def do_task(self, task, round_number):
         func_name = self.task_config[task]['function']
         kwargs = self.task_config[task]['kwargs']
 
+        if (self.device_assignment_policy.name == (
+            'CUDA_PREFERRED' and len(self.cuda_devices) > 0
+        )
+        ):
```

Comment:
If we are using enum, better to use it anywhere
```python
self.device_assignment_policy is DevicePolicy.CUDA_PREFERRED
```
Also it looks like error:
```python
if (self.device_assignment_policy.name == ('CUDA_PREFERRED' and len(self.cuda_devices) > 0)):
```
Maybe
```python
if (self.device_assignment_policy is DevicePolicy.CUDA_PREFERRED 
    and len(self.cuda_devices) > 0):
```

Reply:
Done, thanks